apps/condo/domains/user/integration/passport/utils/user.js

```
@@ -0,0 +1,274 @@
+
+const providerInfoSchema = z.object({
+    name: z.string(),
+    trustEmail: z.boolean(),
+    trustPhone: z.boolean(),
+})
+
+function _ensureUser (foundUser, userType) {
+    if (!foundUser) return
+    // NOTE: hard coded fields used a strong guard!
+    for (const field of ['id', ' type', 'deletedAt', 'name', 'phone', 'isPhoneVerified', 'email', 'isEmailVerified', 'meta']) {
+        if (foundUser.hasOwnProperty(field)) {
```

Biome (1.9.4)

[error] Do not access Object.prototype method 'hasOwnProperty' from target object.

It's recommended using Object.hasOwn() instead of using Object.hasOwnProperty().
See MDN web docs for more details.

(lint/suspicious/noPrototypeBuiltins)